include/rfl/internal/num_fields.hpp

```diff
@@ -97,20 +108,55 @@ struct CountFieldsHelper {
     }
   }
 
+  template <std::size_t n, std::size_t total_arg_num>
+  static consteval bool has_n_base_param() {
+    if constexpr (n > total_arg_num) {
+      return false;
+    } else {
```

Comment:
It's necessary because if `n > total_arg_num`, `std::make_index_sequence<total_arg_num - n>();` will fail  to compile.

So I added this branch to avoid instantiate it.

Reply:
If the `base_param_num` calls only when `index + 1 <= total_arg_num`, then no needed this branch 

```cpp
if constexpr (index == total_arg_num || !has_n_base_param<index + 1, total_arg_num>()) {
 return index;
...
```

Reply:
I found a btter solution:

```cpp
  template <std::size_t n, std::size_t total_arg_num>
  static consteval bool has_n_base_param() {
    constexpr auto right_len = total_arg_num>=n ? total_arg_num-n : 0; // here!
    return []<std::size_t... l, std::size_t... r>(std::index_sequence<l...>,
                                                  std::index_sequence<r...>) {
      return requires { T{any_base<T>(l)..., any(r)...}; };
    }(std::make_index_sequence<n>(), std::make_index_sequence<right_len>());
  }
```